uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/file/CopyPopup.java

```diff
@@ -46,72 +55,62 @@ public void validate( final String value,
-        footer.addButton( CommonConstants.INSTANCE.CopyPopupCreateACopy(),
-                          new Command() {
-                              @Override
-                              public void execute() {
-                                  final String baseFileName = nameTextBox.getText();
-                                  final String originalFileName = path.getFileName();
-                                  final String extension = ( originalFileName.lastIndexOf( "." ) > 0 ? originalFileName.substring( originalFileName.lastIndexOf( "." ) ) : "" );
-                                  final String fileName = baseFileName + extension;
-
-                                  validator.validate( fileName,
-                                                      new ValidatorCallback() {
-                                                          @Override
-                                                          public void onSuccess() {
-                                                              hide();
-                                                              command.execute( new FileNameAndCommitMessage( baseFileName,
-                                                                                                             checkInCommentTextBox.getText() ) );
-                                                          }
-
-                                                          @Override
-                                                          public void onFailure() {
-                                                              Window.alert( CommonConstants.INSTANCE.InvalidFileName0( baseFileName ) );
-                                                          }
-                                                      } );
-                              }
-                          },
-                          IconType.SAVE,
-                          ButtonType.PRIMARY );
-        footer.addButton( CommonConstants.INSTANCE.Cancel(),
-                          new Command() {
-                              @Override
-                              public void execute() {
-                                  hide();
-                              }
-                          },
-                          ButtonType.DEFAULT );
-        add( footer );
+    public CopyPopup( Path path, Validator validator, CommandWithFileNameAndCommitMessage command, View view ) {
+        this.validator = checkNotNull( "validator",
+                                       validator );
+        this.path = checkNotNull( "path",
+                                  path );
+        this.command = checkNotNull( "command",
+                                     command );
+        this.view = checkNotNull( "view",
+                                  view );
+        this.view.getCopyButton().addClickHandler( new ClickHandler() {
+
+            @Override
+            public void onClick( ClickEvent event ) {
+                copy();
```

Comment:
Do the real work in a dedicated method. Separation between view-presenter glue boilerplate improves readability and testability.